src/projectManager.ts

```diff
@@ -359,10 +354,20 @@ export class ProjectManager {
      * @param uri The uri of the location to scan.
      * @param notifyOnNoProjectsFound True to notify the user if no projects were found and false otherwise.
      */
-    public static async scanAndAddSubIProjects(uri: Uri, notifyOnNoProjectsFound: boolean): Promise<void> {
+    public static async scanAndAddSubIProjects(uri: Uri): Promise<void> {
         // Scan for subprojects
         const projectScanDepth = ConfigurationManager.get(ConfigurationSection.projectScanDepth);
-        const scanDepth = projectScanDepth && typeof projectScanDepth === 'number' ? projectScanDepth : 2;
+        const scanDepth = projectScanDepth && typeof projectScanDepth === 'number' ? projectScanDepth : 0;
+        if (scanDepth === 0) {
```

Comment:
Even though the minimum is set to `1` in the `package.json`, we still check if it is set to `0` in case the user manually did this in the `settings.json`